lib/facility.rb

```diff
@@ -1,14 +1,96 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  require 'date'
+  require 'pry'
+  attr_reader :name,
+              :address,
+              :phone,
+              :services,
+              :registered_vehicles,
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_info)
+    @name = facility_info[:name]
+    @address = facility_info[:address]
+    @phone = facility_info[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
-end
+
+  # VEHICLE REGISTRATION
+  def assign_plate_type(vehicle)
+    if vehicle.antique? == true
+      vehicle.plate_type = :antique
+    elsif vehicle.electric_vehicle? == true
+      vehicle.plate_type = :ev
+    else 
+      vehicle.plate_type = :regular
+    end
+  end
+  
+
+  def fees_due(vehicle)
+    if vehicle.plate_type == :antique
+      @collected_fees += 25
+    elsif vehicle.plate_type == :ev
+      @collected_fees += 200
+    else
+      @collected_fees += 100
+    end
+  end
+
+  def register_vehicle(vehicle)
+    if @services.include?("Register Vehicles")
+      vehicle.registration_date = Date.today
+      assign_plate_type(vehicle)
+      
+      @registered_vehicles << vehicle
+      fees_due(vehicle)
+      
+    else
+      print "This Facility Does Not Offer This Service"
+    end
+  end
+
+  #DRIVER'S LICENSE
+
+  #WRITTEN TEST
+
+  def drivers_license_test_prerequisites(registrant)
+    if registrant.permit? == true && registrant.age >= 16
+      true
+    end
```

Comment:
So here, we can refactor this to whole method to be only one line of code that reads `registrant.permit? && registrant.age >=16`, because that statement itself will evaluate to true or false because of the `&&`, so no need to  have an if statement. Nice job encapsulating this logic into a helper method.

Reply:
Also, remember that it is Ruby convention that any method that returns a boolean, should end in a `?`.